modules/widgets/top-posts.php

```diff
@@ -105,6 +130,14 @@ function update( $new_instance, $old_instance ) {
 			$instance['count'] = 10;
 		}
 
+		$allowed_post_types = array_values( get_post_types( array( 'public' => true ) ) );
+		$instance['types'] = $new_instance['types'];
+		foreach( $new_instance['types'] as $type ) {
+			if ( ! in_array( $type, $allowed_post_types ) ) {
+				unset( $type );
```

Comment:
No, I mean if you `unset( $type )`, it won't actually unset the array element, the `$new_instance['types'][ $whatever_the_key_for_that_type ]` will still remain in the array, you need to `unset( $new_instance['types'][ $whatever_the_key_for_that_type ] )` for it to work the way you intended.